aws/aws.go

```diff
@@ -0,0 +1,59 @@
+// aws handles checking Hashicorp Vault issued AWS IAM keys are ready for
+// use. IAM is eventually consistent and if we provide IAM keys to services
+// before they are fully available we will get API errors. Most AWS SDK
+// libraries will fallback to the next available credentials, which in most
+// cases will be the EC2 instance-profile credentials that are available
+// on mesos worker hosts
+package aws
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	RetryLimit = 10
+	RetrySleep = 2 * time.Second
+)
+
```

Comment:
We need to have some tests for this. Which means we will probably want to mock out the calls to AWS. Which means that these `const`s will probably become settings on a thing like `AWSIAMChecker`, which will then have an AWS client dependency injected. And we have an interface that allows us to mock it out in tests. So something like:

*real life*
```go
awsClient := sts.NewFromConfig(cfg)
checker := NewAWSIAMChecker(awsClient)
```

*tests*:
```go
awsClient := &mockSTSClient{}
checker := NewAWSIAMChecker(awsClient)

```